09-intermediate-rails/session.md

```diff
@@ -0,0 +1,117 @@
+  def create
+    @book = Book.create book_params
+
+    if @book.id != nil
+      flash[:success] = "Book added successfully"
+      redirect_to books_path
+    else
+      render :new
+    end
+  end
+```
+
+Now let's test it out to see if this will be displayed somewhere on our book list page after successful creation. Any luck? No!
+
+As with all variables that we set in our controller actions, the way we get this data to display to our users is to put it in a view! Since we can use `flash` in many different contexts, we will add this to our `application.html.erb` view so that any controller action will be able to use this in the future (though you can add this to any view you'd like).
+
+```html
+# app/views/layouts/application.html.erb
+...
+<body>
+  <% flash.each do |name, message| %>
+    <div class="<%= name %>"><%= message %></div>
+  <% end %>
+
+  <%= yield %>
+</body>
+```
+
+This view code will also dynamically add a `class` to the message `div` so that we can add any distinguishing styles to these flash messages.
+
+Now let's see what we end up with!
+
+## `flash.now`
+
+The above example works because right after we set the `flash` value in the controller, we are doing a `redirect_to`. If we want to use a `flash` but with a `render` instead of a `redirect_to` then we need to use `flash.now` instead. The different usage is solely based on what comes next. `flash.now` will only stick around going to a `render` and `flash` will stick around going to a `redirect`.
+
+## `session`
+`session` is another special type of hash-like-object that we utilize to keep track of data throughout a users "session", which normally ends when they close their browser. It is similar to `params` in that it is something special that Rails creates for us. It is different because it persists from one one request to another. It is most often used to store information about a user when they log in.
+
+Rails will automatically delete the information from the `session` when the user closes the browser. We can also manually remove the data if we not longer want to track it (like when the user logs out).
+
+Let's see it! We are going to create a special form which is going to allow our authors to "log in". This is not a secure way to simulate logging in, but we are using this as our first stepping stone to creating "real" authentication, which we'll do later on. We'll allow the user to type an authors name which will "log in" as that author.
+
+We'll need to a create a few new things in our application in order to get set up for this:
+1. Create a login form and form submission route (think of how we do `new` and `create`)
+2. Create a special controller for managing this login process called `SessionsController`
+3. Add empty controller actions for the routes in step #1
+4. Create a form view where our user will enter in the author's name
+5. Create our initial controller action which will set the `session`
+```ruby
+# app/controllers/sessions_controller.rb
```

Comment:
I like this example. Do you have a books repo sitting around that has an author model added to it?